app/webpacker/components/Panel/pages/ApprovePictures/index.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useState } from 'react';
+import {
+  Button,
+  CardGroup,
+  Header,
+  List,
+  Segment,
+} from 'semantic-ui-react';
+import { useMutation } from '@tanstack/react-query';
+import I18n from '../../../../lib/i18n';
+import AvatarCard from './AvatarCard';
+import updateAvatars from './api/updateAvatars';
+import WCAQueryClientProvider from '../../../../lib/providers/WCAQueryClientProvider';
+
+const transformAvatarsForRequest = (approvedAvatars, rejectedAvatars) => {
+  const avatars = {};
+
+  approvedAvatars.forEach(({ id }) => {
+    avatars[id] = { action: 'approve' };
+  });
+
+  rejectedAvatars.forEach(({ id, rejectionReason }) => {
+    avatars[id] = { action: 'reject', rejection_reason: rejectionReason };
+  });
+
+  return { avatars };
+};
+
+export default function Wrapper({ pendingUsers }) {
+  return (
+    <WCAQueryClientProvider>
+      <ApprovePictures pendingUsers={pendingUsers} />
+    </WCAQueryClientProvider>
+  );
+}
+
+function ApprovePictures({ pendingUsers }) {
+  const [approvedAvatars, setApprovedAvatarIds] = useState([]);
+  const [rejectedAvatars, setRejectedAvatarIds] = useState([]);
+  const [deferredAvatars, setDeferredAvatarIds] = useState([]);
+
+  const onApprove = useCallback((avatar) => {
+    setApprovedAvatarIds((prev) => (prev.includes(avatar.id) ? prev.filter((a) => a.id !== avatar.id)
+      : [...prev, avatar]));
+  }, [setApprovedAvatarIds]);
```

Comment:
As far as I can see, `onApprove` can only be called if the avatar is not already approved, so the ternary condition would always be false when this callback is called?
It doesn't really make sense for something called `onApprove` to remove the argument from the approved list, to me.